linalg/sparsemat.cpp

```diff
@@ -660,65 +668,70 @@ void SparseMatrix::AddMult(const Vector &x, Vector &y, const double a) const
 
    // Skip if matrix has no non-zeros
    if (nnz == 0) {return;}
-   if (Device::Allows(Backend::CUDA_MASK) && useCuSparse)
+   if ((Device::Allows(Backend::CUDA_MASK) || Device::Allows(Backend::HIP_MASK)) &&
+       useGPUSparse)
```

Comment:
Combine the mask checks into one: `(Device::Allows(Backend::CUDA_MASK | Backend::HIP_MASK))`.

Reply:
done